packages/services/schema/src/lib/federation-tag-extraction.ts

```diff
@@ -472,10 +353,44 @@ export function applyTagFilterOnSubgraphs<
     name: string;
   },
 >(subgraphs: Array<TType>, filter: Federation2SubgraphDocumentNodeByTagsFilter): Array<TType> {
-  const filteredSubgraphs = subgraphs.map(subgraph => ({
-    ...subgraph,
-    ...applyTagFilterToInaccessibleTransformOnSubgraphSchema(subgraph.typeDefs, filter),
-  }));
+  const specImpl = (resolveName: (name: string) => string) => (subgraph: TType) => {
```

Comment:
I overall find these changes here quite complicated, which makes me doubt whether there is an abstraction that can solve all potential directives. 🤔 I need to digest this a bit more.

Reply:
I understand this concern. I'll take another look at the approach and try and simplify the library and this implementation if possible.